bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/ghost/GhostCB.scala

```diff
@@ -49,12 +52,18 @@ final class GhostCB(obsComp: ISPObsComponent) extends AbstractObsComponentCB(obs
       // must operate on an SPSkyObject.
       def coordParam(so: SPSkyObject, name: Option[String],
                      raDeg: String, decDeg: String,
-                     raHMS: String, decDMS: String) : Unit = so.getCoordinates(None).foreach { c =>
-        name.foreach(n => config.putParameter(systemName, StringParameter.getInstance(n, so.getName)))
-        config.putParameter(systemName, StringParameter.getInstance(raDeg,  c.ra.formatDegrees))
-        config.putParameter(systemName, StringParameter.getInstance(decDeg, c.dec.formatDegrees))
-        config.putParameter(systemName, StringParameter.getInstance(raHMS,  c.ra.formatHMS))
-        config.putParameter(systemName, StringParameter.getInstance(decDMS, c.dec.formatDMS))
+                     raHMS: String, decDMS: String) : Unit = {
+        val coords = so match {
+          case c: SPCoordinates => c.getCoordinates.some
+          case t: SPTarget => t.getProperMotion.map(_.calculateAt(t.getTarget, Instant.now())).orElse(t.getCoordinates(None))
```

Comment:
If we hooked it up to `Target`, which we shouldn't do since it will change existing behavior elsewhere, this would be just `t.coords(instant)`.  As is, it makes sense.

I suppose we could add the time to the parameter list because you sometimes call this for multiple targets and it's better to push this up as much as we easily can anyway.  It will make *no difference* at the rate that these things go but nevertheless ...

It feels odd to not wrap this in `IO` because as is, this isn't referentially transparent, but I guess not in this codebase.